src/providers/oracle/qgsoracledataitems.cpp

```diff
@@ -182,56 +182,46 @@ QVector<QgsDataItem *> QgsOracleConnectionItem::createChildren()
   if ( deferredDelete() )
     return QVector<QgsDataItem *>();
 
-  if ( !mColumnTypeThread )
+  if ( !mColumnTypeTask )
   {
-    mColumnTypeThread = new QgsOracleColumnTypeThread( mName,
+    mColumnTypeTask = new QgsOracleColumnTypeTask( mName,
         QgsOracleConn::restrictToSchema( mName ),
         /* useEstimatedMetadata */ true,
         QgsOracleConn::allowGeometrylessTables( mName ) );
-    mColumnTypeTask = new QgsProxyProgressTask( tr( "Scanning tables for %1" ).arg( mName ) );
-    QgsApplication::taskManager()->addTask( mColumnTypeTask );
 
-    connect( mColumnTypeThread, &QgsOracleColumnTypeThread::setLayerType,
+    connect( mColumnTypeTask, &QgsOracleColumnTypeTask::setLayerType,
              this, &QgsOracleConnectionItem::setLayerType );
-    connect( mColumnTypeThread, &QThread::started, this, &QgsOracleConnectionItem::threadStarted );
-    connect( mColumnTypeThread, &QThread::finished, this, &QgsOracleConnectionItem::threadFinished );
+    connect( mColumnTypeTask, &QgsTask::begun, this, &QgsOracleConnectionItem::taskStarted );
+    connect( mColumnTypeTask, &QgsTask::taskCompleted, this, &QgsOracleConnectionItem::taskFinished );
+    connect( mColumnTypeTask, &QgsTask::taskTerminated, this, &QgsOracleConnectionItem::taskFinished );
 
     if ( QgsOracleRootItem::sMainWindow )
     {
-      connect( mColumnTypeThread, &QgsOracleColumnTypeThread::progress,
-               mColumnTypeTask, [ = ]( int i, int n )
-      {
-        mColumnTypeTask->setProxyProgress( 100.0 * static_cast< double >( i ) / n );
-      } );
-      connect( mColumnTypeThread, SIGNAL( progressMessage( QString ) ),
+      connect( mColumnTypeTask, SIGNAL( progressMessage( QString ) ),
```

Comment:
new style connect please!

Reply:
problem is that the slot `showStatusMessage` is in QgisApp and the provider is not linking with application.

Reply:
Hmm... not sure how I feel about this. If we really want to allow browser items a mechanism for pushing status messages then it would be cleaner to add a signal to QgsDataItem to properly support this (and listen for this signal in the browser model, which connects to all added items and re-emits the signal for connection setup in qgisapp)

Reply:
I fixed this by connecting with QMainWindow::statusBar()